src/core_plugins/vega/public/vega_view/vega_base_view.js

```diff
@@ -188,6 +222,115 @@ export class VegaBaseView {
     }
   }
 
+  /**
+   * Handle
+   * @param funcName
+   * @param args
+   * @returns {Promise<void>}
+   */
+  async vegaFunctionsHandler(funcName, ...args) {
+    try {
+      const handlerFunc = vegaFunctions[funcName];
+      if (!handlerFunc || !this[handlerFunc]) {
+        // in case functions don't match the list above
+        throw new Error(`${funcName}() is not defined for this graph`);
+      }
+      await this[handlerFunc].apply(this, args);
+    } catch (err) {
+      this.onError(err);
+    }
+  }
+
+  /**
+   * @param {object} query Elastic Query DSL snippet, as used in the query DSL editor
+   * @param {string} [index] as defined in Kibana, or default if missing
+   */
+  async addFilterHandler(query, index) {
+    const indexId = await this._findIndex(index);
+    const filter = buildQueryFilter(query, indexId);
+    await this._queryfilter.addFilters(filter);
+  }
+
+  /**
+   * @param {object} query Elastic Query DSL snippet, as used in the query DSL editor
+   * @param {string} [index] as defined in Kibana, or default if missing
+   */
+  async removeFilterHandler(query, index) {
+    const indexId = await this._findIndex(index);
+    const filter = buildQueryFilter(query, indexId);
+
+    // This is a workaround for the https://github.com/elastic/kibana/issues/18863
+    // Once fixed, replace with a direct call (no await is needed because its not async)
+    //    this._queryfilter.removeFilter(filter);
+    this._$rootScope.$evalAsync(() => {
+      try {
+        this._queryfilter.removeFilter(filter);
+      } catch (err) {
+        this.onError(err);
+      }
+    });
+  }
+
+  removeAllFiltersHandler() {
+    this._queryfilter.removeAll();
+  }
+
+  /**
+   * Update dashboard time filter to the new values
+   * @param {number|string|Date} start
+   * @param {number|string|Date} end
+   */
+  setTimeFilterHandler(start, end) {
+    this._timefilter.setTime(VegaBaseView._parseTimeRange(start, end));
+  }
+
+  /**
+   * Parse start and end values, determining the mode, and if order should be reversed
+   * @private
+   */
+  static _parseTimeRange(start, end) {
+    const absStart = moment(start);
+    const absEnd = moment(end);
+    const isValidAbsStart = absStart.isValid();
+    const isValidAbsEnd = absEnd.isValid();
+    let mode = 'absolute';
+    let from;
+    let to;
+    let reverse;
+
+    if (isValidAbsStart && isValidAbsEnd) {
+      // Both are valid absolute dates.
+      from = absStart;
+      to = absEnd;
+      reverse = absStart.isAfter(absEnd);
+    } else {
+      // Try to parse as relative dates too (absolute dates will also be accepted)
+      const startDate = dateMath.parse(start);
+      const endDate = dateMath.parse(end);
+      if (!startDate.isValid() || !endDate.isValid()) {
```

Comment:
`startDate` and `endDate` could be `undefined` in case the user specified an invalid format. I think we should fix that `if` for also checking for that.

Reply:
done. BTW, note that `moment({a:1})` is, for some weird reason, is a valid date :(